pkg/endpoint/events.go

```diff
@@ -119,6 +120,71 @@ func (e *Endpoint) PolicyRevisionBumpEvent(rev uint64) {
 	}
 }
 
+/// EndpointNoTrackEvent contains all fields necessary to update the NOTRACK rules.
+type EndpointNoTrackEvent struct {
+	ep     *Endpoint
+	annoCB AnnotationsResolverCB
+}
+
+// Handle handles the NOTRACK rule update.
+func (ev *EndpointNoTrackEvent) Handle(res chan interface{}) {
+	e := ev.ep
+	port, err := ev.annoCB(e.K8sNamespace, e.K8sPodName)
+	if err != nil {
+		res <- &EndpointRegenerationResult{
+			err: err,
+		}
+		return
+	}
+
+	// If this endpoint is going away, remove all related NOTRACK rules.
+	if err := e.lockAlive(); err != nil {
+		if e.NoTrackPort != "" {
+			if e.IPv4.IsSet() {
+				iptables.RemoveNoTrackRules(e.IPv4.String(), e.NoTrackPort, false)
+			}
+			if e.IPv6.IsSet() {
+				iptables.RemoveNoTrackRules(e.IPv6.String(), e.NoTrackPort, true)
+			}
+			e.NoTrackPort = ""
+		}
+		res <- &EndpointRegenerationResult{
+			err: nil,
+		}
+		return
+	}
+
+	defer func() {
+		e.unlock()
+	}()
+
```

Comment:
[0] This seems like a more appropriate point to perform the input validation of the user-specified annotation, before assigning a (potentially nonsense) value into `e.NoTrackPort`.

Reply:
Added input validation earlier in this function.